nixos/modules/services/networking/nftables.nix

```
@@ -144,7 +150,12 @@ in
           checkPhase = lib.optionalString cfg.checkRuleset ''
             cp $out ruleset.conf
             ${cfg.preCheckRuleset}
-            export NIX_REDIRECTS=/etc/protocols=${pkgs.buildPackages.iana-etc}/etc/protocols:/etc/services=${pkgs.buildPackages.iana-etc}/etc/services
+            export NIX_REDIRECTS=${concatStringsSep ":"
+              [ "/etc/protocols=${pkgs.buildPackages.iana-etc}/etc/protocols"
+                "/etc/services=${pkgs.buildPackages.iana-etc}/etc/services"
+                "/etc/hosts=${config.environment.etc.hosts.source}"
+                "$NIX_REDIRECTS"
+              ]}
```

Use conditional parameter substitution to extend the list with `NIX_REDIRECTS` if and only if it is assigned with a non-zero-length value.
```suggestion
            export NIX_REDIRECTS="${concatStringsSep ":"
              [ "/etc/protocols=${pkgs.buildPackages.iana-etc}/etc/protocols"
                "/etc/services=${pkgs.buildPackages.iana-etc}/etc/services"
                "/etc/hosts=${config.environment.etc.hosts.source}"
              ]}''${NIX_REDIRECTS:+:''${NIX_REDIRECTS}}"
```

---

Nice trick, implemented.